### PR TITLE
feat(DWS): enhance cluster to support apiV2

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^(TestAccResourceDWS_basic|TestAccResourceDWS_basicV2)$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws -v

=== RUN   TestAccResourceDWS_basic
=== PAUSE TestAccResourceDWS_basic
=== RUN   TestAccResourceDWS_basicV2
=== PAUSE TestAccResourceDWS_basicV2
=== CONT  TestAccResourceDWS_basic
=== CONT  TestAccResourceDWS_basicV2
--- PASS: TestAccResourceDWS_basic (2622.37s)
--- PASS: TestAccResourceDWS_basicV2 (2672.24s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws	2672.281s

```

### CODE REVIEW

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 1]
```diff
@@ -42,385 +55,1017 @@ func ResourceDwsCluster() *schema.Resource {
...
+				Required:    true,
+				ForceNew:    true,
+				Description: `Administrator username for logging in to a data warehouse cluster.`,
 			},
-
 			"user_pwd": {
-				Type:      schema.TypeString,
-				Required:  true,
-				Sensitive: true,
+				Type:        schema.TypeString,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:82]: **Sensitive** is missing

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 2]
```diff
@@ -42,385 +55,1017 @@ func ResourceDwsCluster() *schema.Resource {
...
 			"port": {
-				Type:     schema.TypeInt,
-				Computed: true,
-				Optional: true,
-				ForceNew: true,
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     8000,
+				Description: "schema: Required",
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: If the param has a default value, I think we can keep it as optional.

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 3]
```diff
@@ -42,385 +55,1017 @@ func ResourceDwsCluster() *schema.Resource {
...
+				Description: `The enterprise project ID.`,
+			},
+			"kms_key_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The KMS key ID.`,
+			},
+			"kms_key_name": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: can we only use `kms_key_id` to encrypt?

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 4]
```diff
@@ -42,385 +55,1017 @@ func ResourceDwsCluster() *schema.Resource {
...
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `The number of latest manual snapshots that need to be retained when deleting the cluster.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The cluster status.`,
+			},
+			"started_at": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: it's better to use `created`(the previous name) or `created_at`(the common name)

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 5]
```diff
@@ -42,385 +55,1018 @@ func ResourceDwsCluster() *schema.Resource {
...
+		return diag.Errorf("cluster (%s) state is not available to update: %s", d.Id(), err)
 	}
 
-	clusterId := d.Id()
-	// check cluster state is available before update
-	checkErr := checkAndWaitClusterStateAvailable(ctx, client, clusterId, true, d.Timeout(schema.TimeoutUpdate))
-	if checkErr != nil {
-		return diag.Errorf("cluster state is not available to update. cluster_id:%s,error:%s", clusterId, checkErr)
+	expandInstanceStorageChanges := []string{
+		"volume.capacity",
```
Comment (ShiChangkuo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: is it `volume.0.capacity`?

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:82]: done

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: ok

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: yes, we can

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: OK

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: done

**Pull request merged**